nebula3/gclient/net/Connection.py

```diff
@@ -171,6 +171,8 @@ def authenticate(self, user_name, password):
         try:
             resp = self._connection.authenticate(user_name, password)
             if resp.error_code != ErrorCode.SUCCEEDED:
+                self._connection.is_used = False
```

Comment:
the transport is not closed, prefer `self.close()`

Reply:
Is it necessary to close the connection here? I tend to release the connection back to the connection pool and wait for the next authenticate